File: Java/src/main/java/org/codecop/dependencies/parameterise_constructor/MarketingCampaign.java
package org.codecop.dependencies.parameterise_constructor;

import java.time.DayOfWeek;
import java.time.LocalDateTime;

public class MarketingCampaign {

    public boolean isActive() {
        return System.currentTimeMillis() % 2 == 0;
    }

    public boolean isCrazySalesDay() {
        return LocalDateTime.now().getDayOfWeek().compareTo(DayOfWeek.FRIDAY) == 0;
    }
}

File: CSharp/DependencyBreakingKatas/ParameteriseConstructor/MarketingCampaign.cs
using System;

namespace Org.Codecop.Dependencies.ParameteriseConstructor
{
    public class MarketingCampaign
    {
        public bool IsActive()
        {
            return (long) DateTime.Now.TimeOfDay.TotalMilliseconds % 2 == 0;
        }

        public bool IsCrazySalesDay()
        {
            return DateTime.Now.DayOfWeek.Equals(DayOfWeek.Friday);
        }
    }
}
